fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1394,7 +1394,7 @@ bool ClientInfo::canReplace(Reference<ClientInfo> other) const {
 		return true;
 	}
 
-	return !protocolVersion.isCompatible(other->protocolVersion);
+	return !other->protocolVersion.hasStableInterfaces();
```

Comment:
I think what you want here is something a little different. If I remember correctly, the idea behind this function is to say:

1. Two different incompatible protocol versions can replace each other
2. The primary library is favored over external libraries for the exact same version
3. Two compatible versions will favor the larger version, if one is larger

This newly added condition would mean that any old versions can replace each other, even if they are in fact the same version.

If we are actually going to remove the compatible section of the protocol version as is done elsewhere in this PR, then I think this function just becomes:

```
if(protocolVersion == other->protocolVersion && !external) {
    return true;
}

return protocolVersion != other->protocolVersion;
```

Ideally at some point, we would also add logic to always favor newer release versions of the same protocol version, or something like that, but that's a bit orthogonal to this.